File: main/src/main/java/com/sedmelluq/lavaplayer/core/container/ogg/OggMetadata.java
package com.sedmelluq.lavaplayer.core.container.ogg;

import com.sedmelluq.lavaplayer.core.info.track.AudioTrackInfoBuilder;
import com.sedmelluq.lavaplayer.core.info.track.AudioTrackInfoProvider;
import java.util.Collections;
import java.util.Map;

import static com.sedmelluq.lavaplayer.core.info.property.AudioTrackPropertyFactory.coreAuthor;
import static com.sedmelluq.lavaplayer.core.info.property.AudioTrackPropertyFactory.coreLength;
import static com.sedmelluq.lavaplayer.core.info.property.AudioTrackPropertyFactory.coreTitle;

/**
 * Audio track info provider based on OGG metadata map.
 */
public class OggMetadata implements AudioTrackInfoProvider {
  public static final OggMetadata EMPTY = new OggMetadata(Collections.emptyMap(), Long.MAX_VALUE);

  private static final String TITLE_FIELD = "TITLE";
  private static final String ARTIST_FIELD = "ARTIST";

  private final Map<String, String> tags;
  private final Long length;

  /**
   * @param tags Map of OGG metadata with OGG-specific keys.
   */
  public OggMetadata(Map<String, String> tags, Long length) {
    this.tags = tags;
    this.length = length;
  }

  @Override
  public void provideTrackInfo(AudioTrackInfoBuilder builder) {
    builder
        .with(coreTitle(tags.get(TITLE_FIELD)))
        .with(coreAuthor(tags.get(ARTIST_FIELD)));

    if (length != null) {
      builder.with(coreLength(length));
    }
  }
}

File: main/src/main/java/lavaplayer/container/ogg/OggMetadata.kt
package lavaplayer.container.ogg

import lavaplayer.track.info.AudioTrackInfoProvider
import lavaplayer.container.ogg.OggMetadata
import lavaplayer.tools.Units

/**
 * Audio track info provider based on OGG metadata map.
 *
 * @param tags Map of OGG metadata with OGG-specific keys.
 */
class OggMetadata(private val tags: Map<String, String>, override val length: Long?) : AudioTrackInfoProvider {
    companion object {
        @JvmField
        val EMPTY = OggMetadata(emptyMap(), Units.DURATION_MS_UNKNOWN)
        private const val TITLE_FIELD = "TITLE"
        private const val ARTIST_FIELD = "ARTIST"
    }

    override val title: String?
        get() = tags[TITLE_FIELD]

    override val author: String?
        get() = tags[ARTIST_FIELD]

    override val identifier: String?
        get() = null

    override val uri: String?
        get() = null

    override val artworkUrl: String?
        get() = null
}
